torch/lib/c10d/reducer.cpp

```diff
@@ -429,6 +431,69 @@ void Reducer::set_divide_factor() {
   }
 }
 
+void Reducer::delay_all_reduce() {
+  std::lock_guard<std::mutex> lock(this->mutex_);
+
+  if (!static_graph_ || num_iterations_ > 1) {
+    return;
+  }
+
+  if (should_collect_runtime_stats()) {
+    record_backward_compute_end_time();
+  }
+  // The autograd engine uses the default stream when running callbacks, so we
```

Comment:
Can we put this into a helper function since it's also used around `finalize_backward`?